lib/checkautovariables.cpp

```diff
@@ -612,6 +612,8 @@ void CheckAutoVariables::checkVarLifetimeScope(const Token * start, const Token
                         if (tokvalue->exprId() == tok->exprId() && !(tok->variable() && tok->variable()->isArray()) &&
                             !astIsContainerView(tok->astParent()))
                             continue;
+                        if (tokvalue->str() == "=" && Token::simpleMatch(tokvalue->astOperand1(), "."))
```

Comment:
I am not sure exactly what this code does with tokvalue..
should it be checked that the dot is a unary operand? we could have something like `ab.a=...`

Reply:
Or maybe this needs to be fixed in ValueFlow, i.e. the tokvalue should already point at the assigned variable?

Reply:
hmm I am actually not very sure about how LifetimeTokens work

Reply:
The `=` is produced here and then goes unchanged through `LifetimeStore`: https://github.com/danmar/cppcheck/blob/1f35303b920d848415076395999ef72281a46991/lib/valueflow.cpp#L2861
So I guess it's OK to handle the designated initializer later.

Reply:
I see, but that seems like a different problem.
At least there is a FN for this:
~~~c++
struct A { int i; int& r; };
A f() {
    int x = 0;
    A a{.r = x};
    return a;
}
~~~